pallets/asset-index/src/lib.rs

```diff
@@ -677,6 +717,43 @@ pub mod pallet {
 			share.checked_div(&nav).ok_or_else(|| ArithmeticError::Overflow.into())
 		}
 
+		/// Consolidate and clean deposits while completing withdraw
+		fn do_consolidate_deposits(caller: &T::AccountId, mut amount: T::Balance) -> Result<T::Balance, DispatchError> {
+			<Deposits<T>>::try_mutate_exists(&caller, |maybe_depositing| -> Result<T::Balance, DispatchError> {
+				let mut depositing = maybe_depositing.take().ok_or(<Error<T>>::NoDeposits)?;
+				let mut total_fee: T::Balance = Default::default();
+
+				let mut dim: Option<(T::Balance, T::BlockNumber)> = None;
```

Comment:
This is the remainder, right?
can we name this `rem` or `surplus or something like that?